src/lib/modules/ens/embarkjs.js

```diff
@@ -172,36 +172,39 @@ __embarkENS.setProvider = function (config) {
 };
 
 __embarkENS.resolve = function (name, callback) {
-  callback = callback || function () {};
-  if (!this.ens) {
-    return callback(providerNotSetError);
-  }
-  if (!web3.eth.defaultAccount) {
-    return callback(defaultAccountNotSetError);
-  }
-
-  let node = namehash.hash(name);
-
+  callback = callback || function() {};
   function cb(err, addr) {
     if (err === NoDecodeAddrError) {
       return callback(name + " is not registered", "0x");
     }
     callback(err, addr);
   }
 
-  return this.ens.methods.resolver(node).call()
-    .then(resolverAddress => {
+  return new Promise((resolve, reject) => {
+    if (!this.ens) {
+      cb(providerNotSetError);
+      reject(providerNotSetError);
```

Comment:
I would recommend putting the reject inside the `cb()` function and put the `cb()` function inside the Promise, so that way you can check for the `NoDecodeAddrError` error.
Also, that way, you don't have to duplicate the cb and reject for each error.